### DISCUSSION TITLE
position: fixed, em dispositivos mobile

### DISCUSSION THREAD

**Discussion by felipefialho** [frontendbr/forum#314] [Dicas]
Imagino que a maioria já teve esse problema.

O `position: fixed` é completamente ignorado em dispositivos mobile (na verdade, dispositivos touch), e se comporta como um elemento com `position: absolute`.

De tempos em tempos tenho essa questão para resolver em algum projeto, então vou compartilhar com vocês a resolução:

Adicione no HTML:

``` html
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no" />
```

No CSS do elemento com problema (não esqueça do auto prefix):

``` css
backface-visibility hidden 
```

Já pode abrir sua cerveja pra comemorar :)

**Comment by manzettidenis** [comment]
Cara, te amo <3 considere aberta !

**Reply by Mateus-desig** [reply]
Eu amo ele mais ainda !

**Comment by andre-mendes** [comment]
Noossa, tava com esse problema semana passada :D bom saber desse esquema ae \o/

**Comment by felipefialho** [comment]
Acham uma boa escrever um artigo um pouco mais detalhado, com exemplos e tal, no meu site. Ou essa issue já é suficiente pra escalar?

**Comment by fdaciuk** [comment]
@LFeh escreve :heart:

**Comment by eduardojmatos** [comment]
@Lfeh o grande problema é a usabilidade no user-scalable=no, que remove a possibilidade de zoom nos devices :(

**Comment by rogeralbinoi** [comment]
Realmente @eduardojmatos  tbm fiquei meio assim do user-scalable=no :(

**Comment by felipefialho** [comment]
@eduardojmatos, também pensei nisso, mas em tese apenas o `backface-visibility hidden` resolve a boa parte dos casos.

Porém ainda temos problemas com scroll + zoom em alguns dispositivos, e aí não tem jeito. Em tese, se o site manter uma boa proporção de tamanho das fontes e contraste, não teria problema com acessibilidade.

Entrei em um 6 sites para testar e em TODOS o zoom está desativado, inclusive no GitHub. Sinceramente não vejo como um problema muito grande.

Nenhum aplicativo, e nem o próprio sistema operacional do Android, mantém essa opção ativada :p

**Comment by RafaelHGBotelho** [comment]
Muito bom cara, pode escrever sobre isso que vou ler. Sobre o zoom também não vejo problema nenhum, a não ser que seja um site muito minimalista ou que não esteja com a tipografia correta para mobile.

**Comment by talitapagani** [comment]
Será que não tem alguma solução em que não precise utilizar o user-scalable=no? Mesmo com a tipografia de tamanho adequado, creio que há pessoas que ainda assim podem precisar de zoom para enxergar o conteúdo com um tamanho que seja mais confortável para ela. Essa opção acaba retirando o controle do usuário. Será que esse artigo ajuda: http://dbushell.com/2013/09/10/css-fixed-positioning-and-mobile-zoom/ ?

**Comment by felipefialho** [comment]
Então @talitapagani, mas essa é uma parada bem polemica. 

Como disse acima, andei fazendo um teste rápido nos apps que tenho instalado no celular e em diversos sites e todos eles não permitem o zoom. 

Estamos lidando com um problema de acessibilidade de longa escala, sem perceber? haha

**Comment by eduardojmatos** [comment]
@Lfeh então, resolve em partes. O que a @talitapagani falou faz sentido pra mim tbm, de não impedir o zoom, mesmo que tenha uma boa tipografia (tanto em tamanho quanto tipo)

**Comment by eduardojmatos** [comment]
@Lfeh tirando isso, achei sensacional esse jeito de resolver :)

**Comment by felipefialho** [comment]
Alias, essa é uma pauta. Realmente TODOS os apps e páginas que visitei retiraram essa opção. Que loucura.

**Comment by eduardojmatos** [comment]
@Lfeh pode crer, tem vários mesmo :(

**Comment by talitapagani** [comment]
Sim, @LFeh, isso é um problema de acessibilidade de larga escala, bem parecido com aquela questão do outline:none que comentei no FrontinSampa. É como uma _bad design pattern_, rs.

**Comment by felipefialho** [comment]
@talitapagani Movi a discussão pra outra issue, olha lá :D

**Comment by kizylirio** [comment]
Obrigada cara! Funcionou aqui! ^^

**Comment by barraponto** [comment]
Não sei se o @LFeh escreveu em algum lugar, mas vai uma explicação em inglês mesmo que eu achei por aí: https://benfrain.com/improving-css-performance-fixed-position-elements/

**Comment by lucasctd** [comment]
Infelizmente não funcionou comigo. Testei no Edge (Chromium) e no Firefox. :v

**Comment by lucasctd** [comment]
Se alguém puder me ajudar com isso, para reproduzir o problema basta clonar este [repositório ](https://github.com/lucasctd/fathom-modal/tree/support-mobile) e rodar os comandos abaixo:

1. yarn
2. yarn watch/dev
3. yarn start
4. Acessar localhost:8080
5. Utilizar o browser em modo mobile, rolar o scroll até o fim da página e clicar no botão "Show"

**Comment by fdaciuk** [comment]
@lucasctd só tem um "Olá, mundo!" aparecendo.. não tem botão "Show" :)

**Comment by lucasctd** [comment]
@fdaciuk Estranho, tu baixou exatamente o [branch ](https://github.com/lucasctd/fathom-modal/tree/support-mobile)([support-mobile](https://github.com/lucasctd/fathom-modal/tree/support-mobile)) que eu coloquei no link?
Caso a tela não apareça, tente diretamente: http://localhost:8080/index.html

**Comment by fdaciuk** [comment]
Sim, dessa branch mesmo.. agora vi aqui que o `start` inicia a aplicação na porta `8081`, mas ainda assim não é exibido o código que vc está falando.

**EDIT:** Agora consegui acessar na porta `8081/index.html` :)

**Comment by fdaciuk** [comment]
@lucasctd testei sei código aqui, mas nem na versão desktop (com a tela reduzida) ele fica com position: fixed :)

Seu problema está além do mobile xD

**Comment by lucasctd** [comment]
@fdaciuk Entendi, então sua máquina já está utilizando a porta 8080.

**Comment by lucasctd** [comment]
Pode ser algum bug pq modifiquei algumas coisas pra tentar fazer isso funcionar no mobile.
O comportamento real dele é como consta na página abaixo:
https://lucas.mreis.dev/#/portfolio
Passe o mouse sobre um projeto, ao aparecer o overlay, clique em "See details"

**Comment by fdaciuk** [comment]
> @fdaciuk Entendi, então sua máquina já está utilizando a porta 8080.

@lucasctd quando roda o `yarn watch` ele fica ouvindo a porta 8080

**Comment by fdaciuk** [comment]
@lucasctd esse site que vc passou só aparece a opção "See details" na versão desktop.. na versão com tela reduzida ou mobile ele só mostra o link "Visit", não abre a modal :)

**Comment by lucasctd** [comment]
> @lucasctd testei sei código aqui, mas nem na versão desktop (com a tela reduzida) ele fica com position: fixed :)
> 
> Seu problema está além do mobile xD

Sim, só passei esse link pra tu ver o funcionamento correto dele no desktop. Eu desabilitei no _mobile_ pq, como citei acima, o `position fixed` não funciona como esperado.

**Comment by fdaciuk** [comment]
> Sim, só passei esse link pra tu ver o funcionamento correto dele no desktop. Eu desabilitei no mobile pq, como citei acima, o position fixed não funciona como esperado.

Entendi! Então, no exemplo que vc mostrou, ele funciona da forma como está no seu site: no desktop sem rolagem, fica certo. Mas no desktop com rolagem ele não está ficando "fixed". Isso que eu quis dizer :)

**Comment by lucasctd** [comment]
@felipefialho tem alguma ideia do pq não está funcionando?

**Comment by felipefialho** [comment]
@lucasctd Explica melhor, usou a técnica que mostrei lá em cima?

**Comment by lucasctd** [comment]
@felipefialho Não tenho como te explicar o problema agora. Contudo, tenho o mesmo problema em outra aplicação que acredito que tu pode me ajudar.
Neste site: https://designcolumbus.wpengine.com/
Login: demo
Pass: design

Preciso manter o menu fixo no topo, mas o _position fixed_ não funciona, mesmo no modo desktop (no mobile é outro menu).
Sabe me dizer o pq?

**Comment by felipefialho** [comment]
@lucasctd Uhmmm, basta adicionar `transform: scale(1)` no header fixo, que vai funcionar.

**Comment by skreamozz** [comment]
ahi estaba la madre del borrego!! graciasss!!

**Comment by MarcoBosc** [comment]
Cara, salvou minha vida! Obrigado
